qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -513,7 +523,12 @@ def _clean_validate_template(cls, md_template, study_id,
         # In the database, all the column headers are lowercase
         md_template.columns = [c.lower() for c in md_template.columns]
 
-        # Droping/Ignoring internal generated colums
+        # Droping/Ignoring internal generated columns:
+        # A legacy feature of _clean_validate_template() is that it silently
+        # removes these columns before validating. It would be better to make
+        # this optional. These two keywords are also in forbidden words lists.
```

Comment:
Adding here so is in the same block but this is in response to your comment in the tests (*). The reason we have this silent removal is for usability; the normal path is that users add their info files (before they were called templates) without these cols, qiita adds them, the user downloads the new file, adds/modifies columns, uploads. In the past, before these lines, the code will raise an error and will ask the user to remove them, which is less than ideal ...

Reply:
Gotcha. Removed comment, and added behavior description in the docstring, per the conversation above.